makefile: run report_metric post synthesis

oharboe opened the pull request:
It is useful to iterate only on synthesis during early stages of RTL development and in this workflow information on timing is expected.

oharboe committed:
makefile: run report_metric post synthesis

It is useful to iterate only on synthesis during early stages
of RTL development and in this workflow information on timing
is expected.

Signed-off-by: Øyvind Harboe <oyvind.harboe@zylin.com>

maliberty commented on flow/Makefile:
```diff
@@ -401,18 +401,16 @@ ifeq ($(SYNTH_HIERARCHICAL), 1)
 	mkdir -p $(RESULTS_DIR) $(LOG_DIR) $(REPORTS_DIR)
 	($(TIME_CMD) $(YOSYS_CMD) $(YOSYS_FLAGS) -c $(SYNTH_SCRIPT)) 2>&1 | tee $(LOG_DIR)/1_1_yosys.log
+	cp $(SDC_FILE) $(RESULTS_DIR)/1_synth.sdc
+	($(TIME_CMD) $(OPENROAD_CMD) $(SCRIPTS_DIR)/synth_metrics.tcl) 2>&1 | tee -a $(LOG_DIR)/1_1_yosys.log
```
This will add a bunch of runtime for a large design and I'm not sure it will generally be useful.  Perhaps it should be its own target for a specific workflow.

oharboe replied:
how about if I remove the report at the beginning of the floorplan, the  the net change for a full build should be nearly zero.

maliberty replied:
I don't see any report at the beginning of the floorplan.  In any case you would still have to load the design twice.

oharboe replied:
> I don't see any report at the beginning of the floorplan. In any case you would still have to load the design twice.

Ah, towards the end of floorplan:


https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/0f70f74c26b9f697132611f537844f1f6271ded5/flow/scripts/floorplan.tcl#L115

oharboe replied:
How about I move it into the synth target, but not to synthesis?

maliberty replied:
By the end of floorplan we've done remove_buffers which may not give you the metrics you want.  Generally floorplan.tcl is pretty fast so perhaps just add a call to the metrics at the start?  If you really prefer synth, that's ok too.

oharboe committed:
makefile: add do-synth-report target and invoke it from synth target

avoid doing do-synth-report during a normal build, because it
can add a lot of time.

when iterating specifically on the synth target, a timing report
is a more approperiate default behavior

Signed-off-by: Øyvind Harboe <oyvind.harboe@zylin.com>

oharboe committed:
makefile: add a few missing .PHONY statements

Signed-off-by: Øyvind Harboe <oyvind.harboe@zylin.com>

maliberty commented on flow/Makefile:
```diff
@@ -383,9 +383,16 @@ $(WRAPPED_LIBS):
 synth: versions.txt \
        $(RESULTS_DIR)/1_synth.v \
        $(RESULTS_DIR)/1_synth.sdc
+	$(UNSET_AND_MAKE) do-synth-report
```
Same concern about added runtime

oharboe committed:
makefile: add synth-report target

runs synthesis and produces timing report

Signed-off-by: Øyvind Harboe <oyvind.harboe@zylin.com>

oharboe committed:
Merge remote-tracking branch 'origin/master' into HEAD
